### PR TITLE
Allow to specify type for closure arguments

### LINKED ISSUE #574: clone macro: allow to specify a closure's argument type
Like (currently we can't add `: u32`):

```rust
clone!(@weak foo => |x: u32| println!("{}", x));
```

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #574.

cc @sdroege @EPashkin

### CODE REVIEW

**Comment by sdroege** [comment]
> The command "python3 clone_tests/clone_compilation_errors.py" exited with 1.

File: src/clone.rs [hunk 1]
```diff
@@ -144,6 +144,20 @@ macro_rules! to_return_value {
     };
 }
 
+#[doc(hidden)]
+#[macro_export]
+macro_rules! closure_args {
```
Comment (sdroege) [inline src/clone.rs]: You're not using this macro anywhere

**Comment by GuillaumeGomez** [comment]
Oups, forgot to clean up and check...

**Comment by GuillaumeGomez** [comment]
Updated.

**Pull request merged**